app/src/main/java/com/z1/pokedex/feature/home/presentation/screen/HomeScreen.kt

```diff
@@ -190,16 +190,20 @@ fun PokemonList(
     )
 
     val threshold = remember { 5 }
-    val isLastItemVisible by remember {
-        derivedStateOf {
-            if (isShowGridList) {
+    val isLastItemVisible by if (isShowGridList) {
+        remember {
+            derivedStateOf {
                 val layoutInfo =
                     gridListState.layoutInfo
                 val totalItems = layoutInfo.totalItemsCount
                 val lastVisibleItemIndex =
                     (layoutInfo.visibleItemsInfo.lastOrNull()?.index ?: 0) + 1
```

Comment:
Poderia criar uma extension pra um `nullable Int` pra ser o valor dele mesmo ou zero caso nulo, exemplo ->

`fun Int?.orZero() = this ?: 0`

e ao invés do `+ 1` o `.plus(1)`.

Isso é mais perfumaria, acho que soa mais agradável ao ler hehe, mas aí é subjetivo